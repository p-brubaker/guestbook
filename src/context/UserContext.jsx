import { createContext, useState, useContext, useEffect } from 'react';

const UserContext = createContext();

function UserProvider({ children }) {
  const [user, setUser] = useState(null);

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
}

function useUser() {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser hook must be called within a UseProvider');
  }

  return context;
}

export { useUser, UserProvider };
