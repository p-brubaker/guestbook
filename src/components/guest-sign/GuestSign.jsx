import './GuestSign.css';
import { useState } from 'react';
import { useUser } from '../../context/UserContext';

export default function GuestSign({ addGuest }) {
  const { user, setUser } = useUser(null);
  const [entryInput, setEntryInput] = useState('');
  const [nameInput, setNameInput] = useState('');

  async function handleSign() {
    addGuest({ name: nameInput, entry: entryInput });
    setUser(nameInput);
  }

  async function handleChangeName() {
    setUser(null);
  }

  if (user) {
    return (
      <div className="guest-sign-container">
        <p>Thanks for Signing {user}!</p>
        <label htmlFor="guest-entry">Guest Entry</label>
        <input
          name="guest-entry"
          id="guest-entry"
          value={entryInput}
          onChange={(e) => setEntryInput(e.target.value)}
        />
        <button onClick={handleSign}>Sign</button>
        <button onClick={handleChangeName}>Not {user}?</button>
      </div>
    );
  } else {
    return (
      <div className="guest-sign-container">
        <p>Please Sign the Guestbook!</p>
        <label htmlFor="guest-name">Guest Name</label>
        <input
          name="guest-name"
          id="guest-name"
          value={nameInput}
          onChange={(e) => setNameInput(e.target.value)}
        />
        <label htmlFor="guest-entry">Guest Entry</label>
        <input
          name="guest-entry"
          id="guest-entry"
          value={entryInput}
          onChange={(e) => setEntryInput(e.target.value)}
        />
        <button onClick={handleSign}>Sign</button>
      </div>
    );
  }
}
