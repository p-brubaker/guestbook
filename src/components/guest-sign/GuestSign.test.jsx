import { render, screen } from '@testing-library/react';
import GuestSign from './GuestSign';
import React from 'react';
import { UserProvider } from '../../context/UserContext';

it('should render the GuestSign component', async () => {
  const { container } = render(
    <UserProvider>
      <GuestSign />
    </UserProvider>
  );

  screen.getByText('Please Sign the Guestbook!');
  expect(container).toMatchSnapshot();
});
