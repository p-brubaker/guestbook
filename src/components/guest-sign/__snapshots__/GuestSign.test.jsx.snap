// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the GuestSign component 1`] = `
<div>
  <div
    class="guest-sign-container"
  >
    <p>
      Please Sign the Guestbook!
    </p>
    <label
      for="guest-name"
    >
      Guest Name
    </label>
    <input
      id="guest-name"
      name="guest-name"
      value=""
    />
    <label
      for="guest-entry"
    >
      Guest Entry
    </label>
    <input
      id="guest-entry"
      name="guest-entry"
      value=""
    />
    <button>
      Sign
    </button>
  </div>
</div>
`;
